lib/engine/game/g_1822_mx/step/discard_train.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+require_relative '../../g_1822/step/discard_train'
+
+module Engine
+  module Game
+    module G1822MX
+      module Step
+        class DiscardTrain < Engine::Game::G1822::Step::DiscardTrain
+          def trains(corporation)
+            if @game.extra_train_pullman_count(corporation) > 1
+              return corporation.trains.select { |t| @game.extra_train_pullman?(t) }
```

Comment:
Can there be multiple true conditions? Should this be added to super rather than return?

Reply:
Yes.  Added to super.